src/actions/jira/jira.ts

```diff
@@ -1,133 +1,207 @@
 
-    const jira = this.jiraClientFromRequest(request)
+    const resp = new Hub.ActionResponse()
 
+    if (!request.params.state_json) {
+      resp.success = false
+      resp.state = new Hub.ActionState()
+      resp.state.data = "reset"
+      return resp
+    }
+
+    let url
+    if (request.scheduledPlan) {
+      if (request.scheduledPlan.url) {
+        url = request.scheduledPlan.url
+      }
+    }
     const issue = {
-      fields: {
-        project: {
-          id: request.formParams.project,
-        },
-        summary: request.formParams.summary,
-        description: `${request.formParams.description}` +
-          `\nLooker URL: ${request.scheduledPlan && request.scheduledPlan.url}`,
-        issuetype: {
-          id: request.formParams.issueType,
-        },
+      project: {
+        id: request.formParams.project!,
+      },
+      summary: request.formParams.summary,
+      description: request.formParams.description,
+      url,
+      issuetype: {
+        id: request.formParams.issueType!,
       },
     }
-    let response
-    try {
-      await jira.addNewIssue(issue)
-    } catch (e) {
-      response = {success: false, message: e.message}
+
+    const stateJson = JSON.parse(request.params.state_json)
+    if (stateJson.tokens && stateJson.redirect) {
+      try {
+        const client = await this.jiraClient(stateJson.redirect, stateJson.tokens)
+        const newIssue = await client.newIssue(issue)
+        await client.addAttachmentToIssue(newIssue.key, buffer, filename, request.attachment.mime)
+        resp.success = true
+      } catch (e) {
+        resp.success = false
+        resp.message = e.message
+      }
+    } else {
+      resp.success = false
+      resp.state = new Hub.ActionState()
+      resp.state.data = "reset"
     }
-    return new Hub.ActionResponse(response)
+    return new Hub.ActionResponse(resp)
   }
 
   async form(request: Hub.ActionRequest) {
-
     const form = new Hub.ActionForm()
-    try {
-      const jira = this.jiraClientFromRequest(request)
-
-      const [projects, issueTypes] = await Promise.all([
-        jira.listProjects(),
-        jira.listIssueTypes(),
-      ])
-
-      form.fields = [{
-        default: projects[0].id,
-        label: "Project",
-        name: "project",
-        options: projects.map((p: any) => {
-          return {name: p.id, label: p.name}
-        }),
-        type: "select",
-        required: true,
-      }, {
-        label: "Summary",
-        name: "summary",
-        type: "string",
-        required: true,
-      }, {
-        label: "Description",
-        name: "description",
-        type: "textarea",
-        required: true,
-      }, {
-        default: issueTypes[0].id,
-        label: "Issue Type",
-        name: "issueType",
-        type: "select",
-        options: issueTypes
-          .filter((i: any) => i.description)
-          .map((p: any) => {
+    const actionCrypto = new Hub.ActionCrypto()
```

Comment:
shouldn't the oauth link be if the login doesn't work and not all the time?

Reply:
now addressed